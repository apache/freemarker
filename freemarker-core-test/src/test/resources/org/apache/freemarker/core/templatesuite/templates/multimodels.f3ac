<#--
  Licensed to the Apache Software Foundation (ASF) under one
  or more contributor license agreements.  See the NOTICE file
  distributed with this work for additional information
  regarding copyright ownership.  The ASF licenses this file
  to you under the Apache License, Version 2.0 (the
  "License"); you may not use this file except in compliance
  with the License.  You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing,
  software distributed under the License is distributed on an
  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
  KIND, either express or implied.  See the License for the
  specific language governing permissions and limitations
  under the License.
-->
<html>
<head>
<title>FreeMarker: Test of Multiple Model implementations</title>
</head>
<body>

<p>Let's begin with a simple model:</p>
<p>${message}</p>

<p>Cool, now get into the first model. This implements a string, list, and
hash as a single class. Let's try some tests...</p>

<p>${data}</p>

<p>Now as a list...</p>

<#list data as item>${item}<br />
</#list>

<p>Index into a list...</p>
<p>${data[ 1 ]}</p>
<p>List size is: ${data.size}</p>
<p>List size is: ${data["size"]}</p>

<p>Now, again, as a hash. First using dot notation, then using [] notation:</p>

<p>${data.selftest}</p>
<p>${data["selftest"]}</p>

<p>Now for the tricky stuff... use a model to index into another model...</p>
<p>${test}</p>
<p>${data[ test ]}</p>
<p>${self}</p>
<p>${data[ self + "test" ]}</p>

<p>Same thing, this time a List index...</p>
<p>${zero}</p>
<p>${data[ zero ]}</p>
<p>${data[ zero + 1 ]}</p>

<p>Now, do the same recursively...</p>
<p>${data}</p>
<p>${data.model2}</p>
<p>${data.model2( "test" )}</p>
<p>${data.model2( data, data.selftest, message )}</p>

<p>Does this really not work?</p>
<p>${data[ 10 ]}</p>
<p>${data[ 10 ].selftest}</p>
<p>${data[ 10 ].message}</p>

<p>(Again, with Hashes)</p>
<p>${data.nesting1.nested}</p>
<p>${data.nesting1.nested.selftest}</p>

<p>${data["nesting1"].nested}</p>
<p>${data["nesting1"].nested["selftest"]}</p>
<p>${data["nesting1"]["nested"]["selftest"]}</p>

<p>As I suspected! (Manual on Expressions needs updating.)</p>

<p>Second test on list size</p>
<p>${data.one.size}</p>
<p>${data.one["size"]}</p>
</body>
</html>
