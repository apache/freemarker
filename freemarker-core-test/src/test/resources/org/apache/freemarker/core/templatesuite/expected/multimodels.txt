/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
<html>
<head>
<title>FreeMarker: Test of Multiple Model implementations</title>
</head>
<body>

<p>Let's begin with a simple model:</p>
<p>Hello, world!</p>

<p>Cool, now get into the first model. This implements a string, list, and
hash as a single class. Let's try some tests...</p>

<p>MultiModel1 as a string!</p>

<p>Now as a list...</p>

Model1 value: 0<br />
Model1 value: 1<br />
Model1 value: 2<br />
Model1 value: 3<br />
Model1 value: 4<br />
Model1 value: 5<br />
Model1 value: 6<br />
Model1 value: 7<br />
Model1 value: 8<br />
Model1 value: 9<br />
Model3 is alive!<br />

<p>Index into a list...</p>
<p>Model1 value: 1</p>
<p>List size is: Nasty!</p>
<p>List size is: Nasty!</p>

<p>Now, again, as a hash. First using dot notation, then using [] notation:</p>

<p>Selftest of a hash from MultiModel1</p>
<p>Selftest of a hash from MultiModel1</p>

<p>Now for the tricky stuff... use a model to index into another model...</p>
<p>selftest</p>
<p>Selftest of a hash from MultiModel1</p>
<p>self</p>
<p>Selftest of a hash from MultiModel1</p>

<p>Same thing, this time a List index...</p>
<p>0</p>
<p>Model1 value: 0</p>
<p>Model1 value: 1</p>

<p>Now, do the same recursively...</p>
<p>MultiModel1 as a string!</p>
<p>Model2 is alive!</p>
<p>Arguments are:<br />test<br /></p>
<p>Arguments are:<br />MultiModel1 as a string!<br />Selftest of a hash from MultiModel1<br />Hello, world!<br /></p>

<p>Does this really not work?</p>
<p>Model3 is alive!</p>
<p>Selftest from MultiModel3!</p>
<p>Hello world from MultiModel3!</p>

<p>(Again, with Hashes)</p>
<p>Model3 is alive!</p>
<p>Selftest from MultiModel3!</p>

<p>Model3 is alive!</p>
<p>Selftest from MultiModel3!</p>
<p>Selftest from MultiModel3!</p>

<p>As I suspected! (Manual on Expressions needs updating.)</p>

<p>Second test on list size</p>
<p>Key size, not the listSize method.</p>
<p>Key size, not the listSize method.</p>
</body>
</html>
